neanderthal/src/main/kotlin/au/com/outware/neanderthal/data/repository/VariantSharedPreferencesRepository.kt

```diff
@@ -106,4 +126,26 @@ class VariantSharedPreferencesRepository(val klass: Class<out Any>,
             return null
         }
     }
+
+    override fun resetVariants(){
```

Comment:
going one step furth, you could probably inline the getDefaultVariants into here (since this is the only place its used and make this function
```
fun resetVariants(){
    getVariants().forEach { variant ->
        removeVariant(variant.name!!)
    }

    sharedPreferences.getStringSet(DEFAULT_VARIANT_LIST, emptySet()).map { name ->
        Variant(name.removeSuffix(VARIANT_DEFAULT), gson.fromJson(sharedPreferences.getString(name, null), klass))
    }.forEach { variant ->
        addVariant(variant)
    }

    setCurrentVariant(sharedPreferences.getString(DEFAULT_VARIANT))
}
```
This would also require stroing the default variant that is supplied in the constructor